vnext/JSI/Shared/PreparedJavaScriptStore.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+namespace facebook::jsi {
+class PreparedJavaScript;
+}
+
+namespace Microsoft::JSI {
+
+/**
+ * TODO (yicyao)
+ */
+class PreparedJavaScriptStore {
+ public:
+  virtual ~PreparedJavaScriptStore() = 0;
+
+  /**
+   * TODO (yicyao)
+   */
+  virtual std::shared_ptr<const facebook::jsi::PreparedJavaScript> getPreparedJavaScript(
+      const std::string &bundleId) = 0;
+
+  /**
+   * TODO (yicyao)
+   */
+  virtual bool persistPreparedJavaScript(const std::string &bundleId) = 0;
```

Comment:
>bundleId [](start = 60, length = 8)

Could we keep consistent naming? When I looked at this initially, I thought we might have has an ID separate from source URL.

Reply:
Yes I'll change this to sourceUrl later.

---
In reply to: [345465774](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345465774) [](ancestors = 345465774)